theano/sandbox/cuda/dnn.py

```diff
@@ -423,6 +453,40 @@ def make_node(self, img, topgrad, desc, h, w):
         return Apply(self, [img, topgrad, desc, h, w],
                      [CudaNdarrayType(broadcastable)()])
 
+    def infer_shape(self, node, shape):
+        h = shape[0][2]  # Height of input feature maps
+        w = shape[0][3]  # Width of input feature maps
+        kh = shape[1][2]  # Height of each filter
+        kw = shape[1][3]  # Width of each filter
+        out3 = kh
+        out4 = kw
+
+        desc = node.inputs[2].owner.op
+        sh, sw = desc.subsample
+
+        # We don't have the information necessary, namely the weight size so
```

Comment:
That's not actually true: `node.inputs[3]` is weight height and `node.inputs[4]` is weight width